safe_control_gym/controllers/mpc/linear_mpc.py

```diff
@@ -87,8 +87,8 @@ def set_dynamics_func(self):
         dfdxdfdu = self.model.df_func(x=self.X_EQ, u=self.U_EQ)
         dfdx = dfdxdfdu['dfdx'].toarray()
         dfdu = dfdxdfdu['dfdu'].toarray()
-        delta_x = cs.MX.sym('delta_x', self.model.nx, 1)
-        delta_u = cs.MX.sym('delta_u', self.model.nu, 1)
+        delta_x = cs.SX.sym('delta_x', self.model.nx, 1)
```

Comment:
SX builds "scalar" expression graphs which tends to be less memory efficient but faster for computation.
My rule of thumb is that MX should be used in case you have memory problem or if you need black box function evaluation in your casadi expressions. Btw, when you pass `{"expand": True}` as option to casadi `nlpsol` you are transforming all the MX in your NLP to SX.
Also, SX expression are easier to debug.